servicecatalog: Test for launch paths data source fails if using assumed role

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.1
AWS v3.48.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_servicecatalog_launch_paths

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```terraform
data "aws_caller_identity" "current" {}

resource "aws_servicecatalog_principal_portfolio_association" "test" {
  portfolio_id  = aws_servicecatalog_portfolio.test.id
  principal_arn = data. aws_caller_identity.current.arn
}

data "aws_servicecatalog_launch_paths" "test" {
  product_id = aws_servicecatalog_product_portfolio_association.test.product_id

  depends_on = [aws_servicecatalog_principal_portfolio_association.test]
}
```

### Debug Output

```
=== CONT  TestAccAWSServiceCatalogLaunchPathsDataSource_basic
    data_source_aws_servicecatalog_launch_paths_test.go:18: Step 1/1 error: Error running apply: exit status 1
        
        Error: error associating Service Catalog Principal with Portfolio: InvalidParametersException: Invalid parameter principalArn, assumed-role is not a recognized IAM identity type
        
          with aws_servicecatalog_principal_portfolio_association.test,
          on terraform_plugin_test.tf line 80, in resource "aws_servicecatalog_principal_portfolio_association" "test":
          80: resource "aws_servicecatalog_principal_portfolio_association" "test" {
        
--- FAIL: TestAccAWSServiceCatalogLaunchPathsDataSource_basic (65.72s)
```

### Expected Behavior

Should run test.

### Actual Behavior

Test errors.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run='TestAccAWSServiceCatalogLaunchPathsDataSource_basic' -timeout 180m`

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #19957

--- a/aws/data_source_aws_servicecatalog_launch_paths_test.go
+++ b/aws/data_source_aws_servicecatalog_launch_paths_test.go
@@ -90,7 +90,7 @@ resource "aws_servicecatalog_portfolio" "test" {
 }
 
 resource "aws_servicecatalog_product_portfolio_association" "test" {
-  portfolio_id = aws_servicecatalog_portfolio.test.id
+  portfolio_id = aws_servicecatalog_principal_portfolio_association.test.portfolio_id
   product_id   = aws_servicecatalog_product.test.id
 }
 
@@ -129,8 +129,6 @@ func testAccAWSServiceCatalogLaunchPathsDataSourceConfig_basic(rName string) str
 	return composeConfig(testAccAWSServiceCatalogLaunchPathsDataSourceConfig_base(rName), `
 data "aws_servicecatalog_launch_paths" "test" {
   product_id = aws_servicecatalog_product_portfolio_association.test.product_id
-
-  depends_on = [aws_servicecatalog_principal_portfolio_association.test]
 }
 `)
 }

--- a/aws/data_source_aws_servicecatalog_launch_paths_test.go
+++ b/aws/data_source_aws_servicecatalog_launch_paths_test.go
@@ -114,9 +114,13 @@ resource "aws_iam_role" "test" {
 
 data "aws_caller_identity" "current" {}
 
+data "aws_iam_session_context" "current" {
+  arn = data.aws_caller_identity.current.arn
+}
+
 resource "aws_servicecatalog_principal_portfolio_association" "test" {
   portfolio_id  = aws_servicecatalog_portfolio.test.id
-  principal_arn = data.aws_caller_identity.current.arn
+  principal_arn = data.aws_iam_session_context.current.issuer_arn
 }
 `, rName)
 }

